controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -138,12 +163,18 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 	// Check if Status is empty and status annotation is present
 	// Manually restore data.
 	if !r.hostHasStatus(host) {
-		reqLogger.Info("Fetching Status from Annotation")
+		reqLogger.Info("Reconstructing Status from hardwareData and annotation")
 		objStatus, err := r.getHostStatusFromAnnotation(host)
+
 		if err == nil && objStatus != nil {
+			if hardwareData.Name != "" {
+				host.Status.HardwareDetails = hardwareData.Spec.HardwareDetails
+				host.Status = *objStatus
+				reqLogger.Info("updating baremetalHost status with hardwareData")
+			}
 			host.Status = *objStatus
-			if host.Status.LastUpdated.IsZero() {
-				// Ensure the LastUpdated timestamp in set to avoid
+			if err == nil && host.Status.LastUpdated.IsZero() {
```

Comment:
Isn't redundant this nil check, given the previous one?

Reply:
It is indeed, thanks for pointing out. Fixed.